suite-native/fiat-rates/src/fiatRatesMiddleware.ts

```diff
@@ -0,0 +1,46 @@
+import { isAnyOf } from '@reduxjs/toolkit';
+
+import { createMiddlewareWithExtraDeps } from '@suite-common/redux-utils';
+import { transactionsActions, accountsActions, blockchainActions } from '@suite-common/wallet-core';
+
+import { fetchFiatRatesThunk, updateTxsFiatRatesThunk } from './fiatRatesThunks';
+
+export const prepareFiatRatesMiddleware = createMiddlewareWithExtraDeps(
+    (action, { dispatch, extra, next, getState }) => {
+        const {
+            actions: { setWalletSettingsLocalCurrency },
+            selectors: { selectLocalCurrency },
+        } = extra;
+
+        if (isAnyOf(accountsActions.updateAccount, accountsActions.createAccount)(action)) {
+            dispatch(fetchFiatRatesThunk({ rateType: 'current' }));
+            // dispatch(fetchFiatRatesThunk({ rateType: 'lastWeek' }));
```

Comment:
nit: What about these comments? If we only want current fiat rates from account creation, we could delete this. Or are you investigating this?

Reply:
I want to keep them they are related to desktop and I want to move this to common and use for desktop too in close future. I will add some comment above middleware that will explain this.